SignalRClient/Connection.swift

```diff
@@ -79,7 +81,7 @@ public class Connection: SocketConnection {
                 self.transport = transport ?? WebsocketsTransport()
                 self.transport!.delegate = self.transportDelegate
 
-                self.transport!.start(url: self.url)
+                self.transport!.start(url: self.url, headers: self.headers)
```

Comment:
Yes, originally I was thinking to add the headers to the DefaultHttpClient class, within the sendHttpRequest method to add headers for the negotiate request.

But when testing on my SignalR Hub, the only time I received the auth headers (and my Claims Identity populated) was when opening the web socket and the Hub's OnConnectedAsync() was called.

I also saw your answer on StackOverflow saying that the negotiate endpoint is not being used anymore? https://stackoverflow.com/questions/46264982/asp-net-core-2-signalr-1-0-0-alpha2-27025-signalr-negotiate-404-error  -- should we still add the headers to the negotiate request? If so, how does the negotiate mechanism work / how can we test the headers are being sent over the wire?

Also, I see this write-up on the ASP.NET blog https://blogs.msdn.microsoft.com/webdev/2018/02/27/asp-net-core-2-1-0-preview1-getting-started-with-signalr/ and they actually use the [Authorize] attribute for the hub class, but I am guessing that only works due to browser cookies? Whereas for us, iOS clients, will only be able to send the JWT auth header at initial connection and server will act accordingly within the OnConnectedAsync()?

Reply:
Initially there was no `/negotiate` per-se. Instead there was an `OPTIONS` request that served the same purpose. The `OPTIONS` request turned out to be problematic and was changed to a `POST` request to the `/negotiate` path. This request can be skipped for the webSockets transport (although neither this client nor C#/JS clients do this).

SignalR has two layers - HttpConnection (formerly known as SocketConnection) and HubConnection. The HttpConnection is a lower layer that does not really have any framing or protocol. HubConnection is built on top and adds framing and protocol to allow hub method invocation. I don't remember from the top of my head but I think the `Authorize` attribute (because you put it on the Hub class) it only applies to the upper layer. If this is the case, I think it would be quite weird to not require auth for the `negotiate` endpoint but require it for the hub. Even if you don't see negotiate in your code (i.e. there isn't any callback) auth negotiate can (and I think should if the rest of SignalR does) require auth. In this scenario not sending auth headers in the negotiate request means you will not be able to connect to the server at all. 

Now, if you think about other transports (i.e. ServerSentEvents and LongPolling) they are HTTP based so they need an HTTP Client to send requests to the server. Each request will need to send headers. I would like to use the same HTTP Client for this purpose.

(I have not looked at your new revision yet and I may not be able to do this before next week. Sorry!)